File: src/tests/fidl/source_compatibility/addmethod/after.go
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package main

import (
	"context"
	"syscall/zx/fidl"

	lib "fidl/fidl/test/addmethod"
)

type client struct {
	addMethod *lib.ExampleWithCtxInterface
}

func (c client) test() {
	c.addMethod.ExistingMethod(context.Background())
}

type server struct{}

// Assert that server implements the Example interface
var _ lib.ExampleWithCtx = &server{}

func (_ *server) ExistingMethod(fidl.Context) error {
	return nil
}
func (_ *server) NewMethod(fidl.Context) error {
	return nil
}

func main() {}

File: src/tests/fidl/source_compatibility/protocol-method-add/hlcpp/after.cc
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/addmethod/cpp/fidl.h>  // nogncheck
namespace fidl_test = fidl::test::addmethod;

class Impl final : public fidl_test::ExampleProtocol {
  void ExistingMethod() final {}
  void NewMethod() final {}
};

int main(int argc, const char** argv) {
  Impl server{};
  return 0;
}
